From ed93c1f397972bd63e607b5566948df5f4831c20 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Mon, 16 Jul 2018 12:52:20 +0200
Subject: [PATCH] Revert "kconfig: only write '# CONFIG_FOO is not set' for
 visible symbols"

References: bsc#1093118
Patch-mainline: no, suse-specific.

This reverts commit f467c5640c29ad258c3cd8186a776c82fc3b8057.

This breaks config in our x86 builds.

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 scripts/kconfig/symbol.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/scripts/kconfig/symbol.c b/scripts/kconfig/symbol.c
index 7c9a88e91cfa..1b9e8370fe98 100644
--- a/scripts/kconfig/symbol.c
+++ b/scripts/kconfig/symbol.c
@@ -397,10 +397,9 @@ void sym_calc_value(struct symbol *sym)
 			if (!sym_is_choice(sym)) {
 				prop = sym_get_default_prop(sym);
 				if (prop) {
+					sym->flags |= SYMBOL_WRITE;
 					newval.tri = EXPR_AND(expr_calc_value(prop->expr),
 							      prop->visible.tri);
-					if (newval.tri != no)
-						sym->flags |= SYMBOL_WRITE;
 				}
 				if (sym->implied.tri != no) {
 					sym->flags |= SYMBOL_WRITE;
-- 
2.13.7

